src/wallet/coinselection.h

```diff
@@ -73,16 +75,49 @@ struct COutput {
           solvable{solvable},
           safe{safe},
           time{time},
-          from_me{from_me},
-          effective_value{txout.nValue}
-    {}
+          from_me{from_me}
+    {
+        if (feerate) {
+            fee = input_bytes < 0 ? 0 : feerate.value().GetFee(input_bytes);
+            effective_value = txout.nValue - fee.value();
+        }
+    }
+
+    COutput(const COutPoint& outpoint, const CTxOut& txout, int depth, int input_bytes, bool spendable, bool solvable, bool safe, int64_t time, bool from_me, const CAmount fees)
+        : outpoint{outpoint},
+          txout{txout},
+          depth{depth},
+          input_bytes{input_bytes},
+          spendable{spendable},
+          solvable{solvable},
+          safe{safe},
+          time{time},
+          from_me{from_me}
+    {
+        // if input_bytes unknown, the fees should be 0, if input_bytes is known, then the fees should be a positive integer
+        assert((input_bytes < 0 && fees == 0) || (input_bytes > 0 && fees >= 0));
```

Comment:
Optional:
> if input_bytes is known, then the fees should be a positive integer

I see that some of the tests for this constructor pass `fees = 0`. What would need to happen for `fees` to be zero when `input_bytes` is greater than zero? IIRC, when SFFO is being used, we use the actual value of the UTXO instead of its effective value, so there too, a fee greater than zero should follow if both feerate and `input_bytes` are known. So, I wonder here whethere `fees = 0` only occurs in the context of tests, or whether this occurs perhaps also in the context of listing unspents outside of transaction building. If so, it could perhaps be mentioned in the comment here.

Reply:
Yes, `fees = 0` only happens in the context of tests so far. I will mention this here.